examples/jsm/csm/CSM.js

```diff
@@ -0,0 +1,338 @@
+import * as THREE from '../../../build/three.module.js';
+import Frustum from './Frustum.js';
+import FrustumBoundingBox from './FrustumBoundingBox.js';
+import Shader from './Shader.js';
+
+export default class CSM {
+
+	constructor( data ) {
+
+		data = data || {};
+
+		this.camera = data.camera;
+		this.parent = data.parent;
+		this.fov = data.fov || this.camera.fov;
+		this.near = this.camera.near;
+		this.far = data.far || this.camera.far;
+		this.aspect = data.aspect || this.camera.aspect;
```

Comment:
Is there a reason that `fov`, `near`, `far`, and `aspect` are individual options instead of being derived from the camera every time? I'm imagining what happens when the camera aspect or other parameters change, an entirely different camera is used. For example in my application I allow users to switch between a perspective an orthographic camera so it would be nice to have it automagically stay in sync. Maybe an `updateCamera( camera )` or `copyFromCamera( camera )` function would help.

And thank you so much for the addition! I'm looking forward to CSM being integrated back into the project.

Reply:
Well, these options are not derived from camera settings because they don't have to match them. And there's no support for orthographic camera.

Reply:
And unfortunately my implementation have nothing to do with actual integration of CSMs into three.js. Such integration will require a lot of research and changes to be done.